components/cqrs/marshaler_structname.go

```diff
@@ -0,0 +1,49 @@
+package cqrs
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+// StructNameMarshaler uses the struct name (without the package) as the message name.
+//
+// It wraps another marshaler which takes care of the message serialization/deserialization.
+// If none provided, it falls back to JSON marshaler.
+type StructNameMarshaler struct {
+	Marshaler CommandEventMarshaler
```

Comment:
after some thinking, maybe it will be a better idea to not add it as an implementation of Marshaler interface, but like function and add an extra attribute for existing marshallers, like:

```
type JSONMarshaler struct {
	NewUUID func() string
        GenerateName func(v interface{}) string
}
```
with default `GenerateName = ObjectName`.
In that case, you may rename `object.go` to something more meaningful, and keep all names generations methods like functions

In that case, the implementations are still more about output format, and naming is decoupled.

What do you think? :)

Reply:
Sounds good to me. I will update the PR soon